src/components/YourCompany/FolioNumber.vue

```diff
@@ -1,89 +1,262 @@
 
-  @Prop({ default: false })
-  readonly isEditing: boolean
-
-  @Prop({ default: false })
-  readonly invalidSection: boolean
-
-  // Data variable
-  private folioNumber: string = null
+  // Local properties
+  private isEditing = false
+  private formValid = false
+  private folioNumber = ''
+  private dropdown = false
 
   // Validation rules
   private readonly rules: Array<Function> = [
-    (v: string) => /^[0-9A-Za-z]*$/.test(v) || 'Invalid character', // numbers and letters only
-    (v: string) => (!v || v.length <= 50) || 'Cannot exceed 50 characters' // maximum character count
+    (v: string) => (!v || v.length <= 15) || 'Maximum 15 characters reached'
   ]
 
-  /** Called when component is created. */
-  private created (): void {
-    if (this.initialValue) {
-      this.folioNumber = this.initialValue
+  /** Whether folio number has changed from original value. */
+  private get hasFolioNumberChanged (): boolean {
+    const fn = this.folioNumber || null
+    const ov = this.originalValue || null
+    return (fn !== ov)
+  }
+
+  /** When Undo is clicked, restores the folio number to the original value. */
+  private onUndoClicked (): void {
+    this.folioNumber = this.originalValue
+    this.emitNewFolioNumber()
+    this.emitHaveChanges()
+    this.isEditing = false
+  }
+
+  /** When form is submitted, accepts the entered folio number. */
+  private onFormSubmit (): void {
+    if (this.formValid) {
+      this.emitNewFolioNumber()
+      this.emitHaveChanges()
+      this.isEditing = false
     }
   }
 
+  /** When Cancel is clicked, undoes the current change. */
+  private onCancelClicked (): void {
+    this.folioNumber = this.initialValue
+    this.isEditing = false
+  }
+
+  /** Updates local value initially and when prop has changed. */
   @Watch('initialValue', { deep: true, immediate: true })
-  private onFolioNumberPropValueChanged (): void {
+  private onInitialValueChanged (): void {
     this.folioNumber = this.initialValue
   }
 
-  // TODO: never used; delete or keep this?
-  /** When Folio Number changes, emit it to parent. */
-  @Watch('folioNumber')
-  @Emit('folioNumberChange')
-  private emitFolioNumber (val: string): void { }
+  /** Informs parent of new folio number. */
+  @Emit('newFolioNumber')
+  private emitNewFolioNumber (): string {
+    return this.folioNumber
+  }
 
+  /** Informs parent whether we have changes. */
   @Emit('haveChanges')
-  private emitHaveChanges (val: boolean): void {
-    // TODO: implement
+  private emitHaveChanges (): boolean {
+    return this.hasFolioNumberChanged
   }
 
+  /** Informs parent whether we are in editing mode. */
+  @Watch('isEditing')
   @Emit('isEditing')
-  private emitIsEditing (val: boolean): void {
-    // TODO: implement
-  }
+  private emitIsEditing (val: boolean): void {}
 }
 </script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+
+::v-deep .v-input .v-label {
+  font-weight: normal;
+  color: $gray7;
+}
```

Comment:
No action require here, just thoughts. 

We seem to override this often. Do you think one day we should have a base override file for the nested component styles.
ie having a file dedicated to the `v-deep` overrides with potentially a small comment above saying `Vuetify v-text-field label` as an example.

Reply:
Excellent idea!

I like the dedicated file idea too. (We can call it "vuetify-overrides.scss".)

I am not going to worry about this today but, when we have a bit of time, this should help get all Vuetify components in line with our UI designs/wishes.

Reply:
Yeah, I would suggest something like this too. Instead of changing all the places UX demands, we could check with Tracey all the "general look" for basic components and change them. The problem I see doing that now is that it would require a full check and regression tests.

Reply:
Absolutely, down the road i'm sure we will find a minute!